src/sia/services/response_handler.py

```diff
@@ -171,17 +177,33 @@ def process_query(
                 obscore_config=obscore_config,
                 butler_collection=collection,
             )
+
         with capturing_start_span("sia_query") as span:
             span.set_data("query", params)
-            # Execute the query
-            table_as_votable = sia_query(
-                butler,
-                obscore_config,
-                params,
-            )
 
-        # Convert the result to a string
-        result = VotableConverterService(table_as_votable).to_string()
+            try:
+                # Execute the query
+                table_as_votable = sia_query(
+                    butler,
+                    obscore_config,
+                    params,
+                )
+                # Publish success event
+                asyncio.run(
+                    events.sia_query_succeeded.publish(
+                        SIAQuerySucceeded(duration=duration(span))
+                    )
+                )
+            except Exception as e:
+                # Publish failed event
+                asyncio.run(
+                    events.sia_query_failed.publish(
+                        SIAQueryFailed(error=str(e), duration=duration(span))
+                    )
+                )
```

Comment:
Do we want to re-raise the exception here so it bubbles up and (I think) returns a 500 to the client?

Reply:
Yea that's a good point we do want to be re-raising that